kt-review.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+import random
+import sys
+
+def main():
+    """Conduct a knight's tour using a brute force method, representing the
+       board as a dict.
+    """
+
+    if not valid_input():
+        sys.exit("usage: knightstour4.py rows columns attempts")
```

Comment:
Exit doesn't take a message, it takes a status code (here, you'd use 1). Before that, you print the message to stderr.